src/test/scala/chiselTests/ChiselSpec.scala

```diff
@@ -195,6 +195,50 @@ trait ChiselRunners extends Assertions {
   }
 }
 
+trait FileCheck extends BeforeAndAfterEachTestData { this: Suite =>
+  import scala.Console.{withErr, withOut}
+
+  private def sanitize(n: String): String = n.replaceAll(" ", "_").replaceAll("\\W+", "")
+
+  private val testRunDir: os.Path = os.pwd / os.RelPath(BackendCompilationUtilities.TestDirectory)
+  private val suiteDir:   os.Path = testRunDir / sanitize(suiteName)
+  private var checkFile:  Option[os.Path] = None
+
+  override def beforeEach(testData: TestData): Unit = {
+    // TODO check that these are always available
+    val nameDir = suiteDir / sanitize(testData.name)
+    os.makeDir.all(nameDir)
+    checkFile = Some(nameDir / s"${sanitize(testData.text)}.check")
+    super.beforeEach(testData) // To be stackable, must call super.beforeEach
+  }
+
+  override def afterEach(testData: TestData): Unit = {
+    checkFile = None
+    super.afterEach(testData) // To be stackable, must call super.beforeEach
+  }
+
+  /** Elaborate a Module to FIRRTL and check the FIRRTL with FileCheck */
+  def generateFirrtlAndFileCheck(t: => RawModule, fileCheckArgs: String*)(check: String): Unit = {
+    // Filecheck needs the thing to check in a file
+    os.write.over(checkFile.get, check)
+    val result = ChiselStage.emitCHIRRTL(t)
+    val extraArgs = os.Shellable(fileCheckArgs)
+    os.proc("FileCheck", checkFile.get, extraArgs).call(stdin = result)
+  }
```

Comment:
For this and the other method, just a `t: => String` is probably more useful as I can see wanting to use this to check SystemVerilog emission. It isn't too onerous to have a user do `fileCheck(ChiselStage.emitCHIRRTL(new Foo))`.

Reply:
Capturing stdout and stderr is kind of annoying though, maybe I'll provide a 3rd method that just takes a String, and have the other 2 call it?

Reply:
That works!